dio/lib/src/dio_mixin.dart

```diff
@@ -305,28 +293,6 @@ abstract class DioMixin implements Dio {
     );
   }
 
-  @override
-  Future<Response> download(
-    String urlPath,
-    dynamic savePath, {
-    ProgressCallback? onReceiveProgress,
-    Map<String, dynamic>? queryParameters,
-    CancelToken? cancelToken,
-    bool deleteOnError = true,
-    String lengthHeader = Headers.contentLengthHeader,
-    Object? data,
-    Options? options,
-  }) {
-    throw UnsupportedError(
-      'download() is not available in the current environment.',
-    );
-  }
-
-  /// Make http request with options.
-  ///
-  /// [uri] The uri.
-  /// [data] The request data
-  /// [options] The request options.
```

Comment:
@AlexV525
Why is the ‍`download` removed? How to implement!

Reply:
It's been moved to specific adapters with more precise exceptions thrown, and it hasn't actually done anything before.

Filed PR for this: https://github.com/cfug/dio/pull/1916